File: StressTracker5001Server/Models/Attachment.cs
using System.ComponentModel.DataAnnotations;

namespace StressTracker5001Server.Models
{
    public class Attachment
    {
        // Id is a new file name generated when the file is uploaded to storage
        // This helps avoid filename conflicts
        public Guid Id { get; set; }
        public required int CardId { get; set; }
        public Card? Card { get; set; }

        // Original file name
        public required string FileName { get; set; }
        // MIME type of the file
        // e.g., "image/png", "application/pdf"
        public required string ContentType { get; set; }
        public required long FileSize { get; set; }
        public required int UploadedById { get; set; }
        public User? UploadedBy { get; set; }
        public DateTime UploadedAt { get; set; }
    }
}

File: StressTracker5001App/src/schemas/attachment.ts
import { z } from "zod";

export const AttachmentSchema = z.object({
  id: z.uuid(),
  cardId: z.number(),
  fileName: z.string(),
  contentType: z.string(),
  fileSize: z.number(),
  uploadedById: z.number(),
  uploadedAt: z.string().datetime(),
  fileUrl: z.url(),
});
